client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -73,7 +81,7 @@ export default React.createClass( {
 		const sanitizedFieldValues = Object.assign( {}, fieldValues );
 		this.fieldNames.forEach( ( fieldName ) => {
 			if ( typeof fieldValues[ fieldName ] === 'string' ) {
-				sanitizedFieldValues[ fieldName ] = fieldValues[ fieldName ].trim();
+				sanitizedFieldValues[ fieldName ] = deburr( fieldValues[ fieldName ].trim() );
```

Comment:
I've merged it as-is - I completely agree that UX has higher priority, but my point was that this doesn't seem like something that should be fixed in domain-details-form, but in FormStateController. We use this class for handling other forms (probably the WHOIS edit one), so it'd be great if all those forms benefited from this improved UX.